tests/src/App/MappedElement.cpp

```diff
@@ -0,0 +1,143 @@
+    }
+};
+
+TEST_F(MappedElementTest, constructFromNameAndIndex)
+{
+    // Arrange
+    Data::IndexedName indexedName {"EDGE1"};
+    Data::MappedName mappedName {"OTHER_NAME"};
+
+    // Act
+    Data::MappedElement mappedElement {indexedName, mappedName};
+
+    // Assert
+    EXPECT_EQ(mappedElement.index, indexedName);
+    EXPECT_EQ(mappedElement.name, mappedName);
+}
+
+TEST_F(MappedElementTest, moveConstructor)
+{
+    // Arrange
+    auto originalMappedElement = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto originalName = originalMappedElement.name;
+    auto originalIndex = originalMappedElement.index;
+
+    // Act
+    Data::MappedElement newMappedElement {std::move(originalMappedElement)};
+
+    // Assert
+    EXPECT_EQ(originalName, newMappedElement.name);
+    EXPECT_EQ(originalIndex, newMappedElement.index);
+}
+
+TEST_F(MappedElementTest, assignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = mappedElementB;
+
+    // Assert
+    EXPECT_EQ(mappedElementA, mappedElementB);
+}
+
+TEST_F(MappedElementTest, moveAssignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = std::move(mappedElementB);
+
+    // Assert
+    EXPECT_EQ(mappedElementA.name.toString(), "ANOTHER_NAME");
+    EXPECT_EQ(mappedElementA.index.toString(), "EDGE2");
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenNotEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_TRUE(aAndBAreNotEqual);
+    EXPECT_FALSE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE1", "OTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_FALSE(aAndBAreNotEqual);
+    EXPECT_TRUE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, lessThanOperator)
+{
+    // Index is compared first, then mappedName
+
+    // Arrange
+    std::vector<Data::IndexedName> indexedNames {
+        Data::IndexedName {"EDGE1"},
+        Data::IndexedName {"EDGE2"},
+        Data::IndexedName {"EDGE3"},
+    };
+    std::vector<Data::MappedName> mappedNames {
```

Comment:
Does this policy stay for faces and vertices? Is there a possibility they may need to be compared against each other?

Reply:
Also, what about lowercase or other characters?

Reply:
I didn't want to be testing the operators of `IndexedName` and `MappedName`, which are well-tested and known to work, but simply to verify that `mappedName` uses them in the right order.